lib/alchemy/pg_search/search.rb

```diff
@@ -19,18 +19,14 @@ def self.remove_page(page)
       end
 
       ##
-      # index a single page and indexable essences
+      # index a single page and indexable ingredients
       #
       # @param page [Alchemy::Page]
       def self.index_page(page)
         remove_page page
 
         page.update_pg_search_document
-        page.all_elements.includes(:ingredients, contents: :essence).find_each do |element|
-          element.contents.each do |content|
-            content.essence.update_pg_search_document if Alchemy::PgSearch.is_searchable?(content.essence_type)
-          end
-
+        page.all_elements.includes(:ingredients).find_each do |element|
           element.ingredients.each do |ingredient|
             ingredient.update_pg_search_document if Alchemy::PgSearch.is_searchable?(ingredient.type)
           end
```

Comment:
To optimize this we could filter the ingredients before loading all of them

```suggestion
element.ingredients.select { |i| Alchemy::PgSearch.is_searchable?(i.type) }.each do |ingredient|
  ingredient.update_pg_search_document
end
```